include/eve/arch/riscv/top_bits.hpp

```diff
@@ -98,8 +98,8 @@ struct top_bits<Logical>
     {
       // if static_bit_size < 8, we have some dirty bits in resulted unsigned,
       // so cleanning them.
-      u_type mask = detail::set_lower_n_bits<u_type>(static_bits_size);
-      return raw_value & mask;
+      u_type mask = raw_value & detail::set_lower_n_bits<u_type>(static_bits_size);
```

Comment:
Is it about a cast? What's this change?

Reply:
The old code returned an int because `&` promotes. It made a warning because we compared the int to an unsigned in the test. As we kinda want this function to return an unsigned, I moved the part aroudn to prevent the promotion to get out